From 071af20ddcde3b58ddceacfd6eb95b6756af4323 Mon Sep 17 00:00:00 2001
From: Alexander Meshcheryakov <Self-Perfection@users.noreply.github.com>
Date: Sat, 16 Nov 2019 17:53:32 +0300
Subject: [PATCH] Fix linking with Python 3.8

Since Python 3.8 --embed flag needs to be provided to python-config to embed python.
Reference: https://bugs.python.org/issue36721

This was partially addressed in #3170 but only fixed building and not linking with python.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 9fc40b17df..00e1f6a53e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4711,7 +4711,7 @@ if test "$PYTHON_CONFIG" != ""; then
   if test $? -ne 0; then
     with_libpython="no"
   fi
-  LIBPYTHON_LDFLAGS="`${PYTHON_CONFIG} --ldflags`"
+  LIBPYTHON_LDFLAGS="`${PYTHON_CONFIG} --ldflags --embed`" || LIBPYTHON_LDFLAGS="`${PYTHON_CONFIG} --ldflags`"
   if test $? -ne 0; then
     with_libpython="no"
   fi
